src/discord/commands/book.ts

```diff
@@ -0,0 +1,229 @@
+import { CommandInteraction, MessageEmbed, MessageActionRow, MessageSelectMenu, MessageButton, ButtonInteraction, SelectMenuInteraction, Message } from 'discord.js';
+import roomModel, { Room } from '../../models/room.model';
+import sectionModel, { Section } from '../../models/section.model';
+import Timeblock from '../../models/timeblock.model';
+import { DateTime } from 'luxon';
+import { Types } from 'mongoose';
+
+interface TimeblockInformation {
+    startsAt: Date;
+    endsAt: Date;
+    availableCapacity: number;
+}
+
+interface sectionInformation extends Section {
+    _id?: Types.ObjectId;
+}
+
+function dateSuffix(day: number) {
+    if (day > 3 && day < 21) return 'th';
+    switch (day % 10) {
+        case 1:
+            return 'st';
+        case 2:
+            return 'nd';
+        case 3:
+            return 'rd';
+        default:
+            return 'th';
+    }
+}
+
+function getDateOptions(selectedDate?: string) {
+    const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+    const dateOptions = [];
+    const date = new Date();
+
+    for (let i = 0; i < 14; i++) {
+        dateOptions.push({
+            label: `${days[date.getDay()]} - ${months[date.getMonth()]} ${date.getDate()}${dateSuffix(date.getDate())}`,
+            value: date.toLocaleDateString().split('/').reverse().join('-'),
+            default: date.toLocaleDateString().split('/').reverse().join('-') === selectedDate ? true : false,
+        });
+        date.setDate(date.getDate() + 1);
+    }
+    return dateOptions;
```

Comment:
The date option function doesn't seem very readable to me, and using `toLocaleDateString()` can have weird effects depending on where the server ends up hosted (and the date/time settings on the server). Can we refactor some/all of this to use the library [Luxon](https://moment.github.io/luxon/#/?id=luxon)? I think you'll find that working with Luxon is a much nicer experience than vanilla JS Dates.